public/app/features/dashboard/api/publicDashboardApi.ts

```diff
@@ -55,18 +55,39 @@ export const publicDashboardApi = createApi({
       },
       providesTags: ['Config'],
     }),
-    saveConfig: builder.mutation<PublicDashboard, { dashboard: DashboardModel; payload: PublicDashboard }>({
+    createPublicDashboard: builder.mutation<PublicDashboard, { dashboard: DashboardModel; payload: PublicDashboard }>({
       query: (params) => ({
         url: `/uid/${params.dashboard.uid}/public-dashboards`,
         method: 'POST',
         data: params.payload,
       }),
       async onQueryStarted({ dashboard, payload }, { dispatch, queryFulfilled }) {
         const { data } = await queryFulfilled;
-        dispatch(notifyApp(createSuccessNotification('Dashboard sharing configuration saved')));
+        dispatch(notifyApp(createSuccessNotification('Public dashboard created!')));
 
         // Update runtime meta flag
         dashboard.updateMeta({
+          hasPublicDashboard: true,
+          publicDashboardUid: data.uid,
+          publicDashboardEnabled: data.isEnabled,
+        });
+      },
+      invalidatesTags: ['Config'],
+    }),
+    updatePublicDashboard: builder.mutation<PublicDashboard, { dashboard: DashboardModel; payload: PublicDashboard }>({
+      query: (params) => ({
+        url: `/uid/${params.dashboard.uid}/public-dashboards`,
+        method: 'PUT',
+        data: params.payload,
+      }),
+      extraOptions: { maxRetries: 0 },
```

Comment:
```suggestion
      
```

Reply:
change invalidatesTags: ['PublicDashboard'],
(github doesn't let me mark it down)